internal/provider/clickhouse_cluster_resource.go

```diff
@@ -664,7 +704,19 @@ func (m *clickhouseClusterResources) parse(rs *clickhouse.ClusterResources) diag
 			m.Keeper = nil
 		} else {
 			m.Keeper = new(clickhouseClusterResourcesKeeper)
-			m.Keeper.ResourcePresetId = types.StringValue(v.ResourcePresetId)
+			m.Clickhouse.ResourcePresetId = types.StringValue(rs.Clickhouse.ResourcePresetId)
```

Comment:
This block code is Keeper specific. Why do you do manipulations with `ClickHouse.ResourcePresetId`.
Was it a mistake? Did you mean `Keeper.ResourcePresetId` ?

Reply:
As well, please use `v`-object to avoid typos like this.

Reply:
Yeah, looks like a typo.

Reply:
Yes, copy-paste issue